src/view/ClusterDetailsPanel.ts

```diff
@@ -181,17 +181,27 @@ export class ClusterDetailsPanel {
 						// get managed cluster
 						clusterManager = await this._kubeDataLoader.getResources(selectedCluster, "ClusterManager");
 						this._panel.webview.postMessage({"clusterManager":JSON.stringify(clusterManager)});
-					} else {
+					}
+					try {
+						klusterlet = await this._kubeDataLoader.getResources(selectedCluster, "Klusterlet");
+
+					} catch (exceptionVar) {
+						klusterlet = [];
+
+					} 
+					// if this is managed cluster
+					if (klusterlet.length !== 0 ){
+						// get klusterlet
+						this._panel.webview.postMessage({"klusterlet":JSON.stringify(klusterlet)});
 						// get applied manifest work
 						appliedManifestWork = await this._kubeDataLoader.getResources(selectedCluster, "AppliedManifestWork");
 						this._panel.webview.postMessage({"appliedManifestWork":JSON.stringify(appliedManifestWork)});
-						// get klusterlet
-						klusterlet = await this._kubeDataLoader.getResources(selectedCluster, "Klusterlet");
-						this._panel.webview.postMessage({"klusterlet":JSON.stringify(klusterlet)});
 					}
-					// get subscriptions
-					subscriptions = await this._kubeDataLoader.getResources(selectedCluster, "Subscription");
-					this._panel.webview.postMessage({"subscriptions":JSON.stringify(subscriptions)});
+					if (klusterlet.length !== 0 || managedClusters.length !== 0 ){
```

Comment:
Subscription is a hub or managed cluster resources?
I don't think it resides on both

Reply:
Subscription resides in both Hub Cluster and Managed Clusters
in Hub Cluster - subscription status: Propagated
in Managed Cluster - subscription status: Subscribed